src/main/java/de/nplay/moderationbot/config/bot/BotConfig.java

```diff
@@ -0,0 +1,67 @@
+package de.nplay.moderationbot.config.bot;
+
+import de.nplay.moderationbot.config.ConfigService;
+import de.nplay.moderationbot.config.bot.type.BotConfigType;
+import net.dv8tion.jda.api.JDA;
+
+import java.util.Arrays;
+import java.util.Collection;
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+public class BotConfig {
+
+    private final String key;
+    private final String displayName;
+    private final BotConfigType type;
+
+    BotConfig(String key, String displayName, BotConfigType type) {
+        this.key = key;
+        this.displayName = displayName;
+        this.type = type;
+    }
+
+    public String key() {
+        return key;
+    }
+
+    public String displayName() {
+        return displayName;
+    }
+
+    public Optional<String> value() {
+        return ConfigService.get(key);
+    }
```

Comment:
this is bad data flow and error prone, it shouldn't be possible to create a BotConfig instance that isnt mapped to a database value